File: codeforces/27.magnets_[344A]/magnets.py
# Author: José Rodolfo (jric2002)
n = int(input())
number_groups = 1
last_pole = ''
while(n):
  magnet = str(input())
  if (last_pole == magnet[0]):
    number_groups += 1
  last_pole = magnet[1]
  n -= 1
print(number_groups)

File: codeforces/low_difficulty/difficulty_800/27.magnets_[344A]/magnets.cpp
#include <iostream>
#include <cstring>
/* Author: José Rodolfo */
using namespace std;
int main() {
  unsigned int n;
  unsigned int number_groups = 1;
  string magnet;
  char last_pole = ' ';
  cin >> n;
  while (n--) {
    cin >> magnet;
    if (last_pole == magnet[0]) {
      number_groups++;
    }
    last_pole = magnet[1];
  }
  cout << number_groups << endl;
  return 0;
}